presto-main/src/main/java/com/facebook/presto/cost/ConnectorFilterStatsCalculatorService.java

```diff
@@ -53,7 +55,12 @@ public TableStatistics filterStats(
             Map<String, Type> columnTypes)
     {
         PlanNodeStatsEstimate tableStats = toPlanNodeStats(tableStatistics, columnNames, columnTypes);
-        PlanNodeStatsEstimate filteredStats = filterStatsCalculator.filterStats(tableStats, predicate, session);
+        // TODO: Consider re-designing the filter calculator APIs so that a proper Session instance
```

Comment:
It seems like everywhere else, we just blindly cast to `FullConnectorSession`.  Do you know under what circumstances we won't be able to cast?  If not, can we try to just validate that it is a `FullConnectorSession` and just get the session?  I am trying to get rid of these references to `Optional<Session>`, which seem strange, as a session comes with every query.

Reply:
I think currently, it should work all the time since there is only 1 implementation used outside of testing. I wrote this defensively in the case we ever came up with some separate implementation where we couldn't cast it properly. If you're okay with the behavior of failing the query in that case, then I can remove all the `Optional`'s

Reply:
My preference is to fail loudly whenever this presumption no longer holds, rather than to silently default to false.

Reply:
This has been fixed